gym/f110_gym/envs/base_classes.py

```diff
@@ -266,143 +293,169 @@ def update_pose(self, raw_steer, vel):
+                self.params["h"],
+                self.params["m"],
+                self.params["I"],
+                self.params["s_min"],
+                self.params["s_max"],
+                self.params["sv_min"],
+                self.params["sv_max"],
+                self.params["v_switch"],
+                self.params["a_max"],
+                self.params["v_min"],
+                self.params["v_max"],
+            )
 
             # dynamics integration
-            self.state = self.state + self.time_step*(1/6)*(k1 + 2*k2 + 2*k3 + k4)
-        
+            self.state = self.state + self.time_step * (1 / 6) * (
+                k1 + 2 * k2 + 2 * k3 + k4
+            )
+
         elif self.integrator is Integrator.Euler:
             f = vehicle_dynamics_st(
                 self.state,
                 np.array([sv, accl]),
-                self.params['mu'],
-                self.params['C_Sf'],
-                self.params['C_Sr'],
-                self.params['lf'],
-                self.params['lr'],
-                self.params['h'],
-                self.params['m'],
-                self.params['I'],
-                self.params['s_min'],
-                self.params['s_max'],
-                self.params['sv_min'],
-                self.params['sv_max'],
-                self.params['v_switch'],
-                self.params['a_max'],
-                self.params['v_min'],
-                self.params['v_max'])
+                self.params["mu"],
+                self.params["C_Sf"],
+                self.params["C_Sr"],
+                self.params["lf"],
+                self.params["lr"],
+                self.params["h"],
+                self.params["m"],
+                self.params["I"],
+                self.params["s_min"],
+                self.params["s_max"],
+                self.params["sv_min"],
+                self.params["sv_max"],
+                self.params["v_switch"],
+                self.params["a_max"],
+                self.params["v_min"],
+                self.params["v_max"],
+            )
             self.state = self.state + self.time_step * f
-        
+
         else:
-            raise SyntaxError(f"Invalid Integrator Specified. Provided {self.integrator.name}. Please choose RK4 or Euler")
+            raise SyntaxError(
+                f"Invalid Integrator Specified. Provided {self.integrator.name}. Please choose RK4 or Euler"
+            )
 
         # bound yaw angle
-        if self.state[4] > 2*np.pi:
-            self.state[4] = self.state[4] - 2*np.pi
+        if self.state[4] > 2 * np.pi:
+            self.state[4] = self.state[4] - 2 * np.pi
         elif self.state[4] < 0:
-            self.state[4] = self.state[4] + 2*np.pi
+            self.state[4] = self.state[4] + 2 * np.pi
 
         # update scan
-        current_scan = RaceCar.scan_simulator.scan(np.append(self.state[0:2], self.state[4]), self.scan_rng)
+        current_scan = RaceCar.scan_simulator.scan(
```

Comment:
We should bring the update of the scan outside this method.
`update_pose` should be only responsible for updating the racecar state. 

In future, we could compute the scan only if required by the observation space, so it would be better to separate it.